game/test/graphic/DungeonCameraTest.java

```diff
@@ -1,135 +1,230 @@
-     }
-
-     @Test
-     public void test_setFocusPoint() {
-         cam.setFocusPoint(new Point(2, 2));
-         assertNull(cam.getFollowedObject());
-         Mockito.verify(cam).setFocusPoint(any());
-         Mockito.verify(cam).getFollowedObject();
-         Mockito.verifyNoMoreInteractions(cam, pos, frustum, entity);
-     }
-
-     @Test
-     public void test_setFocusPoint_noFollow() {
-         cam_noFollow.setFocusPoint(new Point(2, 2));
-         assertNull(cam_noFollow.getFollowedObject());
-         Mockito.verify(cam_noFollow).setFocusPoint(any());
-         Mockito.verify(cam_noFollow).getFollowedObject();
-         Mockito.verifyNoMoreInteractions(cam_noFollow, pos_noFollow, frustum_noFollow);
-     }
-
-     @Test
-     public void test_isPointInFrustum() {
-         cam.isPointInFrustum(2, 2);
-         Mockito.verify(cam).isPointInFrustum(2, 2);
-         Mockito.verify(frustum).boundsInFrustum(any());
-         Mockito.verifyNoMoreInteractions(cam, pos, frustum, entity);
-     }
-
-     @Test
-     public void test_isPointInFrustum_noFollow() {
-         cam_noFollow.isPointInFrustum(2, 2);
-         Mockito.verify(cam_noFollow).isPointInFrustum(2, 2);
-         Mockito.verify(frustum_noFollow).boundsInFrustum(any());
-         Mockito.verifyNoMoreInteractions(cam_noFollow, pos_noFollow, frustum_noFollow);
-     }
-
-     @Test
-     public void test_getFrustum() {
-         Mockito.verifyNoMoreInteractions(cam, pos, frustum, entity);
-     }
-
-     @Test
-     public void test_getFrustum_noFollow() {
-         Mockito.verifyNoMoreInteractions(cam_noFollow, pos_noFollow, frustum_noFollow);
-     }
-    */
+
+    @BeforeClass
+    public static void setUpGdx() {
+        GdxNativesLoader.load(); // load natives for headless testing
+    }
+
+    @Test
+    public void test_update() {
+        // Prepare
+        Entity entity = new Entity();
+        PositionComponent positionComponent = new PositionComponent(entity, new Point(2, 2));
+        DungeonCamera camera =
+                new DungeonCamera(
+                        positionComponent, Constants.VIEWPORT_WIDTH, Constants.VIEWPORT_HEIGHT);
+
+        // Testing
+        assertEquals(
+                "Pos should be (0, 0, 0) initially.",
+                camera.position,
+                new Vector3(0, 0, 0)); // Camera is initially at (0, 0, 0)
+        camera.update();
+        assertEquals(
+                "Pos should be (2, 2, 0) after update.",
+                camera.position,
+                new Vector3(2, 2, 0)); // Because it follows the positionComponent
+        positionComponent.setPosition(new Point(15, 23));
+        camera.update();
+        assertEquals(
+                "Pos should be (15, 23, 0) after update.",
+                camera.position,
+                new Vector3(15, 23, 0)); // Because it follows the positionComponent
+
+        // Cleanup
+        Game.getEntities().clear();
+        Game.getEntitiesToRemove().clear();
```

Comment:
Den cleanup könnten wir doch auch in ein `@After` stecken, da es eh nach jedem Test ausgeführt wird.
Oder gabs dazu schon ne Entscheidung, wie wir das händeln sollen? Hab das gerade nicht im Kopf, hab gerade auch nix im Wiki gesehen.

Reply:
Ich bin mir nicht sicher... Was ich noch im Kopf habe ist, dass erwähnt wurde das `@Before` und `@After` böse sein 😆